Documentation/Enums/Enum Declarations.md

```diff
@@ -0,0 +1,161 @@
+# Enum Declarations
+
+## Feature Overview
+
+An enumeration, or enum for short, is a static listing of named values.
+Enums are a quick and efficient method of storing a limited range of possibilities as integers.
+
+
+## Commands
+
+### `enum end`
+
+`enum end`
+
+The `enum end` command ends declaring an enum type.
+
+### `enum start`
+
+`enum start` `:` `name`
+
+The `enum start` command starts declaring an enum type.
+
+### `enum value`
+
+`enum value` `:` `name`*`[, value]`*
+
+The `enum value` command declares a value of an enum inside its declaration.
+If provided, the value must be an integer.
+
+
+## Usage
+
+```
+enum start : Direction
+enum value : Unknown 0
+enum value : Horizontal
+enum value : Vertical
+enum end
+```
+
+### CSharp
+
+```
+enum Direction {
+    Unknown = 0;
+    Horizontal;
+    Vertical;
+}
+```
+
+### Java
+
+```
+enum Direction {
+    Unknown = 0;
+    Horizontal;
+    Vertical;
+}
+```
+
+### Python
+
+```
+enum Direction {
+    Unknown = 0;
+    Horizontal;
+    Vertical;
+}
+```
+
+### Ruby
+
+```
+enum Direction {
+    Unknown = 0;
```

Comment:
Stackoverflow thread seemed to not have values at all in one of the examples, not sure if that's exactly what we're looking for though. 

> On Apr 26, 2016, at 10:43 PM, Josh Goldberg <notifications@github.com> wrote:
> 
> In Documentation/Enums/Enum Declarations.md:
> 
> > +
> > +### Python
> > +
> > +```
> > +enum Direction {
> > +    Unknown = 0;
> > +    Horizontal;
> > +    Vertical;
> > +}
> > +```
> > +
> > +### Ruby
> > +
> > +```
> > +enum Direction {
> > +    Unknown = 0;
> This is annoying. As far as I can see there's no simple built-in way to define a series of increasing values in Ruby, so it looks like enum members will have to require a numeric value.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>